File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxConfig.cs
using System;
using JetBrains.Annotations;

namespace Volo.Abp.EventBus.Distributed;

public class OutboxConfig
{
    [NotNull]
    public String Name;//  { get; }

    public Type ImplementationType;// { get; set; }

    public Func<Type, bool> Selector;// { get; set; }

    /**
     * Used to enable/disable sending events from outbox to the message broker.
     * Default: true.
    */
    public boolean IsSendingEnabled;// { get; set; } = true;

    public OutboxConfig(@Nonnull String name)
    {
        Name = Check.NotNullOrWhiteSpace(name, nameof(name));
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/OutboxConfig.java
package com.ueueo.eventbus.distributed;

import lombok.Data;
import org.springframework.lang.NonNull;

import java.util.Objects;
import java.util.function.Function;

@Data
public class OutboxConfig {
    @NonNull
    private String name;

    private Class<?> implementationType;

    private Function<Class<?>, Boolean> selector;

    /**
     * Used to enable/disable sending events from outbox to the message broker.
     * Default: true.
     */
    private boolean isSendingEnabled = true;

    public OutboxConfig(@NonNull String name) {
        Objects.requireNonNull(name);
        this.name = name;
    }
}
